source/common/event/dispatcher_impl.cc

```diff
@@ -165,10 +165,16 @@ void DispatcherImpl::run(RunType type) {
   // event_base_once() before some other event, the other event might get called first.
   runPostCallbacks();
 
-  if (type == RunType::NonBlock) {
-    base_scheduler_.nonBlockingLoop();
-  } else {
-    base_scheduler_.blockingLoop();
+  switch (type) {
+  case RunType::NonBlock:
+    base_scheduler_.runActivatedEvents();
```

Comment:
Thoughts on just passing the enum through to a `run` method on the scheduler? Seems like it would simplify things a bit in intermediate sites?

Reply:
I'm not sure if this simplified anything (moved a switch from one file to another) but I'm fine with it this way. One minor drawback is libevent_scheduler_lib now has to depend on the dispatcher interface, which feels a little awkward to me but doesn't cause any great problems.

Reply:
If you prefer it the other way nbd.